lesson12.md

```diff
@@ -0,0 +1,250 @@
+
+```js
+function myFunction(name) {
+    console.log("Hi " + name);
+}
+let myFunction2 = function(name) {
+    console.log("Hi " + name);
+}
+myFunction("Owen"); // "Hi Owen"
+myFunction2("Harald"); // "Hi Harald"
+```
+
+---
+
+HTML
+
+```html
+<button id="myButton">Click me!</button>
+```
+
+How can we set the `onclick` property from JavaScript?
+
+```js
+let myButtonElement = document.getElementById("myButton");
+myButtonElement.onclick = function() {
+    console.log("My button was clicked");
+}
+```
+
+---
+
+So far, we can get elements from HTML and manipulate them.
+
+* We can set and get properties
+* properties can be numbers, strings, booleans, object, functions...
+* But what if we want to create a new element that's not in HTML yet?
+
+---
+
+<!-- .slide: id="create-element" -->
+
+### document.createElement
+
+Basic usage:
+```js
+let myDiv = document.createElement("div");
+```
+
+However, we need to do a bit more configuration to have something useful.
+
+---
+
+```js
+let myDiv = document.createElement("div"); // 1
+myDiv.textContent = "hello";               // 2
+document.body.appendChild(myDiv);          // 3
+// <div>hello</div> has been added to the page!
+```
+
+1. Create a new HTML element using `document.createElement`. Pass the type of the element (`div`, `button`, `img`, ...) as string.
+2. Set all the properties you like, e.g. `textContent`, `onclick`, ...
+3. `appendChild` add an element to the end of the list of children of a specific parent element. Here, we append it to the document's `<body>`.
+
+---
+
+### .appendChild()
+
+What is the function `appendChild`?
+```js
+document.body.appendChild(myDiv);
+```
+
+Remember our `<body>` tag in the html? `document.body` refers to that element, it is mostly just like any other HTML element. Content within here is what the user sees in their browser
+
+`appendChild` means to "add within this element", and it is the best way to add new elements to the page. `appendChild` can be called on almost any element.
+
+
+---
+
+### Exercise 1: Appending elements
+
+Let's say you have this HTML:
+
+```html
+<div id="myDiv"></div>
+```
+
+How do you turn it to this from JavaScript?
+
+```html
+<div id="myDiv">
+    <span>Hello</span>
+</div>
+```
+
+* Hint: Use `document.getElementById`, `document.createElement`, the `textContent` property and `appendChild`.
+
+---
+
+### Exercise 2: Creating elements in a loop
+
+Use a for loop and createElement to generate the following HTML on your page:
+
+```html
+<div>Element 1</div>
+<div>Element 2</div>
+<div>Element 3</div>
+<div>Element 4</div>
+<div>Element 5</div>
+```
+
+---
+
+### Exercise 3: Create a shopping list app!
+
+It should have one `<input>` field where the user can enter a shopping item. It should have one `<button>`. When the user clicks the button, the text from the `<input>` field should be appended to the shopping list.
+
+![Page](images/shopping.png) <!-- .element width="250px" style="display: block; margin: 0 auto;" -->
```

Comment:
You should add this image to your pull request

Reply:
Done!